godtossing/Pos/Ssc/GodTossing/Core/Types.hs

```diff
@@ -119,18 +128,34 @@ type OpeningsMap = HashMap StakeholderId Opening
 -- | Each node generates several 'SharedSeed's, breaks every
 -- 'SharedSeed' into 'Share's, and sends those encrypted shares to
 -- other nodes (for i-th commitment at i-th element of NonEmpty
--- list). That's exactly what forms 'InnerSharesMap'.
-type InnerSharesMap = HashMap StakeholderId (NonEmpty (AsBinary Share))
+-- list). Then those shares are decrypted.
+type InnerSharesMap = HashMap StakeholderId (NonEmpty (AsBinary DecShare))
 
 -- | In a 'SharesMap', for each node we collect shares which said node
--- has received and decrypted. Specifically, if node identified by
--- 'StakeholderId' X has received NonEmpty list of shares from node
--- identified by key Y, this NonEmpty list will be at @sharesMap ! X !
--- Y@.
+-- has received and decrypted:
+--
+--   * Outer key = who decrypted the share
+--   * Inner key = who created the share
+--
+-- Let's say that there are participants {A, B, C}. If A has generated a
+-- secret and shared it, A's shares will be denoted as Ab and Ac (sent
+-- correspondingly to B and C). Then node B will decrypt the share and get
```

Comment:
Did you change the logic so that A doesn't generate shares for A?

Reply:
It looks like I haven't. Oh well, I'll update the comment.

Reply:
I think that we already decrypt our own shares when we process our message, but I'm not 100% sure.